src/ethereum/prague/fork.py

```diff
@@ -113,44 +118,56 @@ def apply_fork(old: BlockChain) -> BlockChain:
     return old
 
 
-def get_last_256_block_hashes(chain: BlockChain) -> List[Hash32]:
+def process_block_hash_history(chain: BlockChain, block: Block) -> None:
     """
-    Obtain the list of hashes of the previous 256 blocks in order of
-    increasing block number.
-
-    This function will return less hashes for the first 256 blocks.
-
-    The ``BLOCKHASH`` opcode needs to access the latest hashes on the chain,
-    therefore this function retrieves them.
+    Update the block hash history before block execution.
+    Introduced in EIP-2935. https://eips.ethereum.org/EIPS/eip-2935
 
     Parameters
     ----------
     chain :
         History and current state.
-
-    Returns
-    -------
-    recent_block_hashes : `List[Hash32]`
-        Hashes of the recent 256 blocks in order of increasing block number.
+    block :
+        Block to be executed.
     """
-    recent_blocks = chain.blocks[-255:]
-    # TODO: This function has not been tested rigorously
-    if len(recent_blocks) == 0:
-        return []
-
-    recent_block_hashes = []
-
-    for block in recent_blocks:
-        prev_block_hash = block.header.parent_hash
-        recent_block_hashes.append(prev_block_hash)
-
-    # We are computing the hash only for the most recent block and not for
-    # the rest of the blocks as they have successors which have the hash of
-    # the current block as parent hash.
-    most_recent_block_hash = keccak256(rlp.encode(recent_blocks[-1].header))
-    recent_block_hashes.append(most_recent_block_hash)
+    # This function should not be called on pre-fork block
+    # since it determines whether the current block is the fork block
+    # by looking at the parent block's timestamp.
+    # The following ensures that the block is not a pre-fork block.
```

Comment:
```suggestion
    # Calling this function on blocks from previous forks is a bug
    # since it determines whether `block` is the fork block
    # by looking at the parent block's timestamp.
    #
    # The following ensures that `block` belongs to this fork.
```

Played around with the wording a bit. Take it or leave it :P

Reply:
Done